### PR TITLE
:seedling: ci: bump Flatcar tested version

### PR DESCRIPTION
**Pull request by tormath1**
**What this PR does / why we need it**: Test a [new](https://www.flatcar.org/releases#release-3602.2.0) Flatcar major Stable

**Special notes for your reviewer**: As discussed previously, let's bump Flatcar tested version only for new major. I followed these instructions: https://gist.github.com/tormath1/acbae5c6cd12420bb8ea137e25655c99 (if the tests are green one maintainer could upload the Flatcar image to the CAPO GCS bucket :pray: , thanks!)  

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: hack/ci/cloud-init/controller.yaml.tpl [hunk 1]
```diff
@@ -138,7 +138,7 @@
     # https://docs.openstack.org/glance/latest/admin/quotas.html
     /opt/stack/devstack/tools/upload_image.sh https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/cirros/2022-12-05/cirros-0.6.1-x86_64-disk.img
     /opt/stack/devstack/tools/upload_image.sh https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/ubuntu/2023-01-14/focal-server-cloudimg-amd64.img
-    /opt/stack/devstack/tools/upload_image.sh https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/flatcar/flatcar-stable-3510.2.4-kube-v1.27.2.img
+    /opt/stack/devstack/tools/upload_image.sh https://storage.googleapis.com/tormath1-flatcar/flatcar-stable-3602.2.0-kube-v1.27.2.img
```
Comment (wwentland) [inline hack/ci/cloud-init/controller.yaml.tpl:141]: Is there any way in which we wouldn't have to switch between `artifacts.k8s-staging-capi-openstack.appspot.com/test` and `tormath1-flatcar` whenever a new flatcar image is being made available?

I appreciate that an automatic process would be best (cf. https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1502), but maybe we can find a different solution in the interim?

**Review by wwentland — COMMENTED** [review]
Thank you, looking forward to testing the current stable version. 

FWIW, I've used that version in my own clusters already and did not run into issues.

File: hack/ci/cloud-init/controller.yaml.tpl (hunk 1 above)
Comment (wwentland) [reply hack/ci/cloud-init/controller.yaml.tpl:141]: > Yes, it's not ideal and it creates bottleneck: one needs to generate the image with the image-builder and one with CAPO GCS Bucket write access needs to upload it.

Indeed, that's exactly what I was wondering we could achieve here: Find a person with the permissions above or grant them.

> One different approach would be to remove the image-builder dependency and to use plain Flatcar image: [...] In the Flatcar template, using Ignition, we could pull Kubernetes dependencies using [systemd-sysext](https://www.flatcar.org/docs/latest/container-runtimes/getting-started-with-kubernetes/#setup-the-control-plane):

I am *very* much looking forward to utilising that functionality in all my clusters and, quite likely, in the context of tests here. How far has that feature progressed by now? My impression was that it is still quite experimental. But, if we can make it work, I'd very much like that. Until then, my preference would be for tests to mirror the setup that most users would use.

On that note: Please give me a shout if you need any help in getting the templates ready.

Comment (mdbooth) [reply hack/ci/cloud-init/controller.yaml.tpl:141]: I still think the ideal situation here is to generate them directly in the CI. Incidentally, the 'OpenStack' packer builder is vastly quicker and more reliable than the qemu builder. The only problem with it is you need an OpenStack to run it against. That's hurting us.

**Comment by mdbooth** [comment]
@tormath1 I've copied the new image to our storage bucket. I'm happy to merge this once you've bumped the location. Thanks!

/approve

**Review by wwentland — COMMENTED** [review]
That's great, thanks again!

/lgtm

**Review by wwentland — COMMENTED** [review]
Cheers!

/lgtm

**Pull request merged**